pkg/chain/ethereum/ethutil/ethutil.go

```diff
@@ -213,3 +213,35 @@ func (lw *loggingWrapper) EstimateGas(ctx context.Context, msg ethereum.CallMsg)
 func WrapCallLogging(logger log.EventLogger, client EthereumClient) EthereumClient {
 	return &loggingWrapper{client, logger}
 }
+
+// BlockHeightWaiter provides the ability to wait for a given block height.
+type BlockHeightWaiter interface {
+	WaitForBlockHeight(blockNumber uint64) error
+}
+
+// WaitForChainConfirmation ensures that after receiving specific number of block
+// confirmations the state of the chain is actually as expected. It waits for
+// predefined number of blocks since the start block number provided. After the
+// required block number is reached it performs a check of the chain state with
+// a provided function returning a boolean value.
+func WaitForChainConfirmation(
+	blockHeightWaiter BlockHeightWaiter,
+	startBlockNumber uint64,
+	blockConfirmations uint64,
+	stateCheck func() (bool, error),
+) (bool, error) {
+	blockHeight := startBlockNumber + blockConfirmations
+	logger.Infof("waiting for block [%d] to confirm chain state", blockHeight)
+
+	err := blockHeightWaiter.WaitForBlockHeight(blockHeight)
+	if err != nil {
+		return false, fmt.Errorf("failed to wait for block height: [%v]", err)
+	}
+
+	result, err := stateCheck()
+	if err != nil {
+		return false, fmt.Errorf("failed to get chain state confirmation: [%v]", err)
+	}
+
+	return result, nil
+}
```

Comment:
What do you think about putting the new code in a separate file, e.g. `pkg/chain/ethereum/ethutil/confirmations.go`? Everything that is inside `ethutil.go` is Ethereum-specific and this code is not. We could even have it outside of `ethutil` package one day. (* I'd stick with `ethutil` package for now and revisit later.)